pkg/scheduler/plugins/elasticquota/plugin.go

```diff
@@ -0,0 +1,560 @@
+
+	if elasticQuotaInfo != nil {
+		if err := elasticQuotaInfo.addPodIfNotPresent(podInfoToAdd.Pod); err != nil {
+			klog.ErrorS(err, "Failed to add Pod to its associated elasticQuota", "pod", klog.KObj(podInfoToAdd.Pod))
+		}
+	}
+
+	return framework.NewStatus(framework.Success, "")
+}
+
+// RemovePod is called by the framework while trying to evaluate the impact
+// of removing podToRemove from the node while scheduling podToSchedule.
+func (g *Plugin) RemovePod(ctx context.Context, state *framework.CycleState, podToSchedule *corev1.Pod,
+	podInfoToRemove *framework.PodInfo, nodeInfo *framework.NodeInfo) *framework.Status {
+	elasticQuotaSnapshotState, err := getElasticQuotaSnapshotState(state)
+	if err != nil {
+		klog.ErrorS(err, "Failed to read elasticQuotaSnapshot from cycleState", "elasticQuotaSnapshotKey", SnapshotStateKey)
+		return framework.NewStatus(framework.Error, err.Error())
+	}
+
+	quotaName := g.getPodAssociateQuotaName(podInfoToRemove.Pod)
+	elasticQuotaInfo := elasticQuotaSnapshotState.elasticQuotaInfos[quotaName]
+
+	if elasticQuotaInfo != nil {
+		if err := elasticQuotaInfo.deletePodIfPresent(podInfoToRemove.Pod); err != nil {
+			klog.Infof("Failed to delete Pod from its associated elasticQuota:%v, pod:%v,err:%v", quotaName, klog.KObj(podInfoToRemove.Pod), err)
+		}
+	}
+
+	return framework.NewStatus(framework.Success, "")
+}
+
+// PostFilter modify the defaultPreemption, only allow pods in the same quota can preempt others.
+func (g *Plugin) PostFilter(ctx context.Context, state *framework.CycleState, pod *corev1.Pod,
+	filteredNodeStatusMap framework.NodeToStatusMap) (*framework.PostFilterResult, *framework.Status) {
+	nnn, status := g.preempt(ctx, state, pod, filteredNodeStatusMap)
+	if !status.IsSuccess() {
+		return nil, status
+	}
+	// This happens when the pod is not eligible for preemption or extenders filtered all candidates.
+	if nnn == "" {
+		return nil, framework.NewStatus(framework.Unschedulable)
+	}
+
+	return &framework.PostFilterResult{NominatedNodeName: nnn}, framework.NewStatus(framework.Success)
+}
+
+func (g *Plugin) Reserve(ctx context.Context, state *framework.CycleState, p *corev1.Pod, nodeName string) *framework.Status {
+	g.Lock()
+	defer g.Unlock()
+
+	quotaName := g.getPodAssociateQuotaName(p)
+	// update groupQuotaManager
+	quotaInfo := g.groupQuotaManager.GetQuotaInfoByName(quotaName)
+	podRequest, _ := resource.PodRequestsAndLimits(p)
+	g.groupQuotaManager.UpdateGroupDeltaUsed(quotaInfo.Name, podRequest)
+	// update elasticQuotaInfos
+	elasticQuotaInfo := g.elasticQuotaInfos[quotaName]
+	if elasticQuotaInfo != nil {
+		err := elasticQuotaInfo.addPodIfNotPresent(p)
+		if err != nil {
+			klog.ErrorS(err, "Failed to add Pod to its associated elasticQuota", "pod", klog.KObj(p))
+			return framework.NewStatus(framework.Error, err.Error())
+		}
+	}
+	// update quotaOverUsedRevokeController
+	g.quotaOverUsedRevokeController.addPodWithQuotaName(quotaName, p)
```

Comment:
g.quotaOverUsedRevokeController.addPodWithQuotaName should be executed at OnPodAdd\Update, not in Reserve

Reply:
fixed